string direction
int16 number
-----------------
string text

